### ISSUE TITLE
Cortex-M0/M0+ systems do not start on release

### ISSUE THREAD

**Issue by ilg-ul** [micro-os-plus/cmsis-plus#8]
With GCC 5.4, any optimisation level higher then `-O1` triggers a hard fault during startup.

**Comment by ilg-ul** [comment]
The problem was specific to Cortex-M0/M0+ configurations. The startup sequence needs to get the MSP stack address, but ARM v6M cores do not have the VTOR register, so the address needs to be fetched from address 0x0.

Intended as a feature, with optimisation level higher then `-O1`, recent GCC versions check if the NULL pointer is dereferenced, and generate an undefined instruction, so instructions like `__set_MSP (*((uint32_t*) 0x0));` are no longer possible.

The current workaround was to use a volatile pointer:

```
uint32_t* volatile vectors_addr = 0x00000000;
 __set_MSP (*vectors_addr);
```

Please note that this currently works on GCC 5.4, on any optimisation level; it is not guaranteed to work on future GCC versions, which might get smarter in checking NULL dereferencing.

**Comment by ilg-ul** [comment]
fixed on v6.3.11.
